Refactored Quarkus tutorial

ammbra opened the pull request:
1. mvn replaced ./mwnw as that command does not work regardless of OS
2. health contains a check to the external service and a description about the default health checks
3. security sections were moved and checked
4. reactive section was changed to use only the api.punkapi.com
5. more examples on reactive streaming
6.  code for the Dockerfile used in docker-compose was updated.

@lordofthejars can you please help me to build quay.io/rhdevelopers/quarkus-tutorial-price-generator:2.0

sebastienblanc commented on apps/price-generator/src/main/resources/application.properties:
```diff
@@ -1,14 +1,16 @@
+mp.messaging.outgoing.priced-beer.topic=priced-beer
+mp.messaging.outgoing.priced-beer.value.serializer=org.apache.kafka.common.serialization.StringSerializer
+
+kafka.auto.offset.reset=earliest
```
Why do we need this property ?

ammbra replied:
By configuring `auto.offset.reset` to `earliest` the application binds to a topic and can consume the historical messages present in a topic. The default is set to `latest`, meaning the application receives the messages that arrived to the topic after it subscribed to the topic. I can remove it and keep the default.

sebastienblanc replied:
Since we are ticking every second, there will always be new records available. I'm not sure this is needed and could add confusion. `earliest` is often used when we need "replayability" behaviour

ammbra replied:
I removed it to avoid confusion.

sebastienblanc commented on documentation/modules/ROOT/pages/basics.adoc:
```diff
@@ -7,7 +7,7 @@
 [source,bash,subs="+macros,+attributes"]
 ----
-mvn "io.quarkus:quarkus-maven-plugin:{quarkus-version}:create" \
+mvn "io.quarkus:quarkus-maven-plugin:create" \
```
Is the version not needed ? Nice, didn't knew that

ammbra replied:
Is not needed as it will create automatically the project with latest released version. I preferred to do this instead of using the Quarkus CLI to have some control over the  version used.

sebastienblanc commented on documentation/modules/ROOT/pages/reactive-messaging.adoc:
```diff
@@ -0,0 +1,365 @@
+    @Incoming("groups")
+    @Outgoing("messages")
+    @Blocking
+    public String processGroup(List<String> list) {
```
What the idea of this method ? Just to show that Mutiny can handle blocking logic as well ? Shouldn't we rename the method to something like `processGroupBlocking` ?

ammbra replied:
It is more to showcase that when you have a blocking operation within your method's logic, you can signal that with Blocking annotation. In this case I am not using Mutiny.

sebastienblanc replied:
make sense

sebastienblanc commented on documentation/modules/ROOT/pages/reactive.adoc:
```diff
@@ -2,7 +2,9 @@
 
-In this chapter we're going to see some examples of how Mutiny changes the design of our Quarkus applications. We're going to develop an asynchronous REST client that connects to an online beer database (https://punkapi.com/documentation/v2) to retrieve beer information. This API does not return all beers at once, so we'll need to navigate through the pages to fetch all the information. Then we'r ...
+In this chapter we're going to see some examples of how Mutiny changes the design of our Quarkus applications. 
+to https://www.fruityvice.com/ to retrieve fruit information based on their carbohydrates. 
```
Some words are missing here I believe

ammbra replied:
Yes, initially I tried to reuse the fruityvice API but it did not serve the purpose well.

sebastienblanc commented on documentation/modules/ROOT/pages/reactive.adoc:
```diff
@@ -2,7 +2,9 @@
-In this chapter we're going to see some examples of how Mutiny changes the design of our Quarkus applications. We're going to develop an asynchronous REST client that connects to an online beer database (https://punkapi.com/documentation/v2) to retrieve beer information. This API does not return all beers at once, so we'll need to navigate through the pages to fetch all the information. Then we'r ...
+In this chapter we're going to see some examples of how Mutiny changes the design of our Quarkus applications. 
+to https://www.fruityvice.com/ to retrieve fruit information based on their carbohydrates. 
+This API does not return all beers at once, so we'll need to navigate through the pages to fetch all the information. Then we're going to filter all the beers with an ABV greater than 15.0 and return these values in a Reactive REST endpoint.
```
`return all the beers`

sebastienblanc commented on documentation/modules/ROOT/pages/security-oidc.adoc:
```diff
@@ -26,18 +26,19 @@ Simply go to http://localhost:8080/q/dev/[Dev UI] and select the OpenId Connect
 Click on the http://localhost:8080/q/dev/io.quarkus.quarkus-oidc/provider[Provider: Keycloak link] and you will see a Keycloak page which will be presented slightly differently depending on how Dev Services for Keycloak feature has been configured.
 
-TIP: By default, `alice` and `bob` users (with the passwords matching the names), and user and admin roles are created. `alice` has both admin and user roles and `bob` has the user role.
+TIP: By default, `alice` and `bob` users (with the passwords matching the names), and user and admin roles are created. `alice` has both admin and user roles and `bob` - the user role.
```
Why `-` ? Maybe `has just` ?

ammbra replied:
Great suggestion :) Thank you!

sebastienblanc commented on documentation/modules/ROOT/pages/spring.adoc:
```diff
@@ -32,6 +32,11 @@ Quarkus supports many of your favorite Spring Framework projects. In this chapte
 
+IMPORTANT: The extensions above require `quarkus-resteasy-jackson` or `quarkus-resteasy-reactive-jackson`.
+In order to be conservative with our dependencies, please change in `pom.xml` the extension `quarkus-reasteasy` 
+to `quarkus-resteasy-jackson`.
```
Not sure I understand the remark here

ammbra replied:
When we start our tutorial the project is generated with `quarkus-reasteasy` . However, when adding the Spring extensions you will notice an error message in the Dev Mode saying to you that they work with either `quarkus-resteasy-jackson` or `quarkus-resteasy-reactive-jackson`. This is why added this line here: to change in `pom.xml` from `quarkus-reasteasy` to `quarkus-resteasy-jackson`.

sebastienblanc replied:
Indeed, I just see it while testing ... Well that's an "interesting" feature that we had not before ... Isn't this a regression we should report to Quarkus ?

ammbra commented:
@sebastienblanc I pushed a new commit addressing some of your comments. Please review the changes done and discuss those still not addressed. Thank you! 

Also, would it make sense to create a portion with resteasy reactives and/or using reactive panache?

sebastienblanc commented:
@ammbra BTW, any idea why there are old commits in this PR (commits from august) ?

ammbra commented:
> 

Yes, because I forgot to switch to master branch in my local fork :(
